src/context.js

```diff
@@ -190,6 +192,29 @@ function EvolvContext() {
     ensureInitialized();
     return key in remoteContext || key in localContext;
   };
+
+  /**
+   * Adds value to specified array in context. If array doesnt exist its created and added to.
+   *
+   * @param key The array to add to.
+   * @param value Value to add to the array.
+   * @param local {Boolean} If true, the value will only be added to the localContext.
+   * @param limit {Number} Max length of array to maintain.
+   * @returns {boolean} True if value was successfully added.
+   */
+  this.pushToArray = function(key, value, local, limit) {
+    limit = limit || DEFAULT_QUEUE_LIMIT;
+
+    ensureInitialized();
+
+    const context = local ? localContext : remoteContext;
+    const originalArray = objects.getValueForKey(key, context);
+
+    const combined = (originalArray || []).concat([value]);
+    const newArray = combined.slice(combined.length - limit);
```

Comment:
The current code is creating 3 new arrays each time. I think the following would be more efficient as it would only create one for immutability. 

`
const newArray = originalArray ? originalArray.splice(-limit).push(value) : [value];
`

Reply:
`push` doesnt return an array unfortunately